whirlpool/dryer.py

```diff
@@ -0,0 +1,281 @@
+LOGGER = logging.getLogger(__name__)
+
+# Machine State
+ATTR_MACHINE_STATE = "Cavity_CycleStatusMachineState"
+
+ATTRVAL_MACHINE_STATE_STANDBY = "0"
+ATTRVAL_MACHINE_STATE_SETTING = "1"
+ATTRVAL_MACHINE_STATE_DELAY_COUNT_DOWN_MODE = "2"
+ATTRVAL_MACHINE_STATE_DELAY_PAUSE = "3"
+ATTRVAL_MACHINE_STATE_SMART_DELAY = "4"
+ATTRVAL_MACHINE_STATE_SMART_GRID_PAUSE = "5"
+ATTRVAL_MACHINE_STATE_PAUSE = "6"
+ATTRVAL_MACHINE_STATE_RUNNING_MAIN_CYCLE = "7"
+ATTRVAL_MACHINE_STATE_RUNNING_POST_CYCLE = "8"
+ATTRVAL_MACHINE_STATE_EXCEPTIONS = "9"
+ATTRVAL_MACHINE_STATE_COMPLETE = "10"
+ATTRVAL_MACHINE_STATE_POWER_FAILURE = "11"
+ATTRVAL_MACHINE_STATE_SERVICE_DIAGNOSTIC = "12"
+ATTRVAL_MACHINE_STATE_FACTORY_DIAGNOSTIC = "13"
+ATTRVAL_MACHINE_STATE_LIFE_TEST = "14"
+ATTRVAL_MACHINE_STATE_CUSTOMER_FOCUS_MODE = "15"
+ATTRVAL_MACHINE_STATE_DEMO_MODE = "16"
+ATTRVAL_MACHINE_STATE_HARD_STOP_OR_ERROR = "17"
+ATTRVAL_MACHINE_STATE_SYSTEM_INIT = "18"
+ATTRVAL_MACHINE_STATE_CANCELLED = "19"
+
+# Cavity
+ATTR_DOOR_OPEN = "Cavity_OpStatusDoorOpen"
+ATTR_EST_TIME_REMAINING = "Cavity_TimeStatusEstTimeRemaining"
+ATTR_DRUM_LIGHT_ON = "Cavity_DisplaySetDrumLightOn"
+
+# DryCavity
+ATTR_STATUS_EXTRA_POWER_CHANGEABLE = "Cavity_ChangeStatusExtraPowerChangeable"
+ATTR_STATUS_STEAM_CHANGEABLE = "Cavity_ChangeStatusSteamChangeable"
+ATTR_STATUS_CYCLE_SELECT = "DryCavity_ChangeStatusCycleSelect"
+ATTR_STATUS_DRYNESS = "DryCavity_ChangeStatusDryness"
+ATTR_STATUS_MANUAL_DRY_TIME = "DryCavity_ChangeStatusManualDryTime"
+ATTR_STATUS_STATIC_GUARD = "DryCavity_ChangeStatusStaticGuard"
+ATTR_STATUS_TEMPERATURE = "DryCavity_ChangeStatusTemperature"
+ATTR_STATUS_WRINKLE_SHIELD = "DryCavity_ChangeStatusWrinkleShield"
+
+ATTR_CYCLE_SELECT = "DryCavity_CycleSetCycleSelect"
+
+ATTRVAL_CYCLE_SELECT_REGULAR = 1
+ATTRVAL_CYCLE_SELECT_HEAVY_DUTY = 2
+ATTRVAL_CYCLE_SELECT_WHITES = 16
+ATTRVAL_CYCLE_SELECT_TOWELS = 15
+ATTRVAL_CYCLE_SELECT_BULKY_ITEMS = 6
+ATTRVAL_CYCLE_SELECT_QUICK_DRY = 7    
+ATTRVAL_CYCLE_SELECT_DELICATES = 4
+ATTRVAL_CYCLE_SELECT_WRINKLE_CONTROL = 5
+ATTRVAL_CYCLE_SELECT_NORMAL = 41
+ATTRVAL_CYCLE_SELECT_TIMED_DRY = 11
+
+ATTR_DRYNESS = "DryCavity_CycleSetDryness"
+
+ATTRVAL_DRYNESS_LOW = 1
+ATTRVAL_DRYNESS_NORMAL = 4
+ATTRVAL_DRYNESS_HIGH = 7
+
+ATTR_MANUAL_DRY_TIME = "DryCavity_CycleSetManualDryTime"
+ATTR_WRINKLE_SHIELD = "DryCavity_CycleSetWrinkleShield"
+
+ATTR_TEMPERATURE = "DryCavity_CycleSetTemperature"
+
+ATTRVAL_TEMPERATURE_LOW = 2
+ATTRVAL_TEMPERATURE_MEDIUM = 5
+ATTRVAL_TEMPERATURE_MEDIUM_HIGH = 6
+ATTRVAL_TEMPERATURE_HIGH = 8
+
+ATTR_AIR_FLOW_STATUS = "DryCavity_CycleStatusAirFlowStatus"
+ATTR_COOL_DOWN = "DryCavity_CycleStatusCoolDown"
+ATTR_DAMP = "DryCavity_CycleStatusDamp"
+ATTR_DRYING = "DryCavity_CycleStatusDrying"
+ATTR_LIMITED_CYCLE = "DryCavity_CycleStatusLimitedCycle"
+ATTR_SENSING = "DryCavity_CycleStatusSensing"
+ATTR_STATIC_REDUCE = "DryCavity_CycleStatusStaticReduce"
+ATTR_STEAMING = "DryCavity_CycleStatusSteaming"
+ATTR_WET = "DryCavity_CycleStatusWet"
```

Comment:
how are these shown in the app UI? For the current implementation in HA, it seems that they are like "sub-states" of ATTRVAL_MACHINE_STATE_RUNNING_MAIN_CYCLE. Is that the case?
I am wondering if/how we could make the name a bit more clear, but without knowing how whirlpool shows them on the app, its hard :sweat_smile:

Reply:
I believe it shows the status with a line of text below the current cycle.  I think the "Drying" at the top is simply saying that the dryer is running.  To be honest, I haven't really looked that much at it since all we every use is Timed Dry without any frills. :-)  And, if you watch the status messages come through, they are pretty quick so it's doubtful you'd ever see them in the app.  (Could be wrong though.)

![cycle_status](https://github.com/user-attachments/assets/52a5dd5b-530f-41f2-8dce-d557269b8854)

Reply:
Maybe lets then prefix them with `ATTR_CYCLE_STATUS_` (and also the respective methods).

sorry if that was how it was initially.